### PR TITLE
feat (p2p): melt addresses inserted through addPeers

### PR DESCRIPTION
**Pull request by girazoki**
This PR adds methods to remove addresses from the ice bucket, and removes those addresses added manually thtough addPeers method

### CODE REVIEW

File: node/src/actors/peers_manager/handlers.rs [hunk 1]
```diff
@@ -14,6 +14,9 @@ impl Handler<AddPeers> for PeersManager {
     fn handle(&mut self, msg: AddPeers, _: &mut Context<Self>) -> Self::Result {
         // Insert address
         log::trace!("Adding the following peer addresses: {:?}", msg.addresses);
+        if msg.src_address == None {
```
Comment (tmpolaczyk) [inline node/src/actors/peers_manager/handlers.rs:17]: You can add a comment explaining why do we need this:

    // Remove peers added manually from the ice bucket to ensure that they are always added
    // The easiest way to check if a peer was added manually is by using msg.src_address,
    // which is set to None when using the addPeers JSON-RPC method and also when reading peers from config

Also, clippy prefers `if msg.src_address.is_none()`

File: p2p/src/peers/mod.rs [hunk 2]
```diff
@@ -374,6 +374,21 @@ impl Peers {
...
     }
+
+    /// Remove address from the ice bucket
+    pub fn ice_peer_address_remove(&mut self, address: &SocketAddr) {
+        log::trace!("Removing peer address {} from the ice bucket", address);
+
+        self.ice_bucket.remove(address);
+    }
+
+    /// Remove a list of addresses from the ice bucket
```
Comment (tmpolaczyk) [inline p2p/src/peers/mod.rs:385]: Mention that this does not take into account the ice_period, and always removes the peers

```suggestion
    /// Remove a list of addresses from the ice bucket, ignoring the ice_period
```

File: p2p/src/peers/mod.rs [hunk 3]
```diff
@@ -374,6 +374,21 @@ impl Peers {
 
         true
     }
+
+    /// Remove address from the ice bucket
+    pub fn ice_peer_address_remove(&mut self, address: &SocketAddr) {
```
Comment (aesedepece) [inline p2p/src/peers/mod.rs:379]: For uniformity with the other buckets:
```suggestion
    pub fn remove_from_ice(&mut self, address: &SocketAddr) {
```

File: p2p/src/peers/mod.rs [hunk 4]
```diff
@@ -374,6 +374,21 @@ impl Peers {
...
+        log::trace!("Removing peer address {} from the ice bucket", address);
+
+        self.ice_bucket.remove(address);
+    }
+
+    /// Remove a list of addresses from the ice bucket
+    pub fn ice_peer_addresses_remove(&mut self, addresses: &[SocketAddr]) {
```
Comment (aesedepece) [inline p2p/src/peers/mod.rs:386]:
```suggestion
    pub fn remove_many_from_ice(&mut self, addresses: &[SocketAddr]) {
```

File: p2p/src/peers/mod.rs [hunk 5]
```diff
@@ -241,6 +241,11 @@ impl Peers {
         addrs
             .iter()
             .filter_map(|address| {
+                // Remove from iced
```
Comment (tmpolaczyk) [inline p2p/src/peers/mod.rs:244]: No, this is "Add to iced"

Comment (girazoki) [reply p2p/src/peers/mod.rs:244]: damn it

File: p2p/src/peers/mod.rs [hunk 6]
```diff
@@ -389,6 +388,21 @@ impl Peers {
...
+    /// Remove address from the ice bucket, ignoring the ice period
+    pub fn remove_from_ice(&mut self, address: &SocketAddr) {
+        log::trace!("Removing peer address {} from the ice bucket", address);
+
+        self.ice_bucket.remove(address);
+    }
+
+    /// Remove a list of addresses from the ice bucket, ignoring the ice_period
+    pub fn remove_many_from_ice(&mut self, addresses: &[SocketAddr]) {
+        log::trace!("Removing peer address {:?} from the ice bucket", addresses);
```
Comment (aesedepece) [inline p2p/src/peers/mod.rs:401]:
```suggestion
        log::trace!("Removing the following peer addresses from the ice bucket: {:?} ", addresses);
```

File: p2p/src/peers/mod.rs [hunk 7]
```diff
@@ -256,6 +256,11 @@ impl Peers {
         addrs
             .iter()
             .filter_map(|address| {
+                // Add to iced
+                if ice {
+                    self.ice_peer_address(address);
+                }
```
Comment (tmpolaczyk) [inline p2p/src/peers/mod.rs:262]: Can we add a test for this change? Check that when trying to remove an address that did not exist, this address will be iced.

Comment (girazoki) [reply p2p/src/peers/mod.rs:262]: sure

**Pull request merged**